src/edit_session_test.js

```diff
@@ -1125,7 +1125,17 @@ module.exports = {
             }
         });
         var session = new EditSession([]);
-        session.setMode("ace/mode/javascript");
+        
+        var onChangeModeCallCount = 0;
+        var originalOnChangeMode = session.$onChangeMode;
+
+        // Create spy
+        session.$onChangeMode = function(...arguments) {
```

Comment:
Why do we need the spy here we already could see the call by effects it produces